cmd/controller/sparsefilegenerator.go

```diff
@@ -0,0 +1,180 @@
+//  file directory on the node.
+func GetSparseFileDir() string {
+
+	sparseFileDir := os.Getenv(ENV_SPARSE_FILE_DIR)
+
+	if len(sparseFileDir) < 1 {
+		return ""
+	}
+
+	info, err := os.Stat(sparseFileDir)
+	if os.IsNotExist(err) || !info.Mode().IsDir() {
+		glog.Info("Specified directory doesnt exist:  ", sparseFileDir)
+		return ""
+	}
+
+	return sparseFileDir
+}
+
+// GetSparseFileCount returns the number of sparse files to be
+//  created by NDM. Returns 0, if invalid count is specified.
+func GetSparseFileCount() int {
+
+	sparseFileCountStr := os.Getenv(ENV_SPARSE_FILE_COUNT)
+
+	if len(sparseFileCountStr) < 1 {
+		sparseFileCountStr = SPARSE_FILE_DEFAULT_COUNT
+	}
+
+	sparseFileCount, econv := strconv.Atoi(sparseFileCountStr)
+	if econv != nil {
+		glog.Info("Error converting sparse file count:  ", sparseFileCountStr)
+		return 0
+	}
+
+	return sparseFileCount
+}
+
+// GetSparseFileSize returns the size of the sparse file to be
+//  created by NDM. Returns 0, if invalid size is specified.
+func GetSparseFileSize() int64 {
+
+	sparseFileSizeStr := os.Getenv(ENV_SPARSE_FILE_SIZE)
+	if len(sparseFileSizeStr) < 1 {
+		sparseFileSizeStr = SPARSE_FILE_DEFAULT_SIZE
+	}
+
+	sparseFileSize, econv := strconv.ParseInt(sparseFileSizeStr, 10, 64)
+	if econv != nil {
+		glog.Info("Error converting sparse file size:  ", sparseFileSizeStr)
+		return 0
+	}
+
+	return sparseFileSize
+}
+
+// InitializeSparseFiles will check if the sparse file exist or have to be
+//  created and will update or create the associated Disk CR accordingly
+func (c *Controller) InitializeSparseFiles() {
+	sparseFileDir := GetSparseFileDir()
+	sparseFileSize := GetSparseFileSize()
+	sparseFileCount := GetSparseFileCount()
+
+	if len(sparseFileDir) < 1 || sparseFileSize < 1 || sparseFileCount < 1 {
+		glog.Info("No sparse file path/size provided. Skip creating sparse files.")
+		return
+	}
+
+	for i := 0; i < sparseFileCount; i++ {
+		sparseFile := path.Join(sparseFileDir, fmt.Sprint(i)+"-"+SPARSE_FILE_NAME)
+		err := CheckAndCreateSparseFile(sparseFile, sparseFileSize)
+		if err != nil {
+			glog.Info("Error creating sparse file: ", sparseFile, "Error: ", err)
+			continue
+		}
+		c.MarkSparseDiskStateActive(sparseFile, sparseFileSize)
+	}
+}
+
+// CheckAndCreateSparseFile will reuse the existing sparse file if it already exists,
+//   for handling cases where NDM is upgraded or restarted. If the file doesn't exist
+//   a new file will be created.
+func CheckAndCreateSparseFile(sparseFile string, sparseFileSize int64) error {
+	sparseFileInfo, err := util.SparseFileInfo(sparseFile)
+	if err != nil {
+		err = util.SparseFileCreate(sparseFile, sparseFileSize)
+	} else {
+		glog.Info("Sparse file already exists: ", sparseFileInfo.Name())
+	}
+	return err
+}
+
+// GetSparseDiskUuid returns a fixed UUID for the sparse
+//  disk on a given node.
+func GetSparseDiskUuid(hostname string, sparseFile string) string {
+	return SPARSE_DISK_PREFIX + util.Hash(hostname+sparseFile)
```

Comment:
why we are changing prefix as we are adding disk type label is it not a duplicate entry or is it for visual only.

Reply:
For ease of use only @shovanmaity